### PR TITLE
moar goodies

### PR DESCRIPTION
**Pull request by vadimplobanov**

### CODE REVIEW

File: libs/freertos.mk [hunk 1]
```diff
@@ -4,41 +4,39 @@
...
 # Note that this implementation does not allow free
+ifeq ($(FREERTOS_HEAP),heap_1)
+  FREERTOS_WARNING = "Memory cannot be freed (user-selected)."
+endif
 ifeq ($(FREERTOS_HEAP),)
-$(warning No heap selected, defaulting to heap 1 (Memory cannot be freed!!))
-FREERTOS_HEAP = heap_1
+  FREERTOS_WARNING = "Memory cannot be freed (default)."
+  FREERTOS_HEAP = heap_1
 endif
```
Comment (Jeff-Ciesielski) [inline libs/freertos.mk]: Nice, I like the distinction

File: libs/freertos.mk [hunk 2]
```diff
@@ -4,41 +4,39 @@
...
-FREERTOS_OBJS = $(FREERTOS_SRCS:.c=.o)
+FREERTOS_C_FILES += $(FREERTOS_SOURCE)/portable/GCC/$(FREERTOS_PORT)/port.c
+FREERTOS_C_FILES += $(FREERTOS_SOURCE)/portable/MemMang/$(FREERTOS_HEAP).c
+FREERTOS_C_FILES += $(sort $(wildcard $(FREERTOS_SOURCE)/*.c))
 
 #FreeRTOS Includes
-LIB_INCLUDES += -I$(FREERTOS_SOURCE)/include \
-	-I$(FREERTOS_SOURCE)/portable/GCC/$(FREERTOS_PORT)
+LIB_INCLUDES += -I$(FREERTOS_SOURCE)/include
+LIB_INCLUDES += -I$(FREERTOS_SOURCE)/portable/GCC/$(FREERTOS_PORT)
```
Comment (Jeff-Ciesielski) [inline libs/freertos.mk]: Any particular reason to ditch the slash and include multiple lines?  I have more often seen the former than the latter.  Not opposed to it, just curious.

Comment (vadimplobanov) [inline libs/freertos.mk]: No technical reason -- was purely a stylistic choice. Do you prefer the other way around?

Comment (Jeff-Ciesielski) [inline libs/freertos.mk]: No, I'm happy to have it as is, if it gets to be a crazy long list, I guess we'll see how it looks in regard to the other parts of the build system.

**Pull request merged**